src/webui/webapplication.cpp

```diff
@@ -700,11 +710,46 @@ bool WebApplication::validateHostHeader(const QStringList &domains) const
     }
 
     LogMsg(tr("WebUI: Invalid Host header. Request source IP: '%1'. Received Host header: '%2'")
-           .arg(m_env.clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
+           .arg(m_clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
             , Log::WARNING);
     return false;
 }
 
+QHostAddress WebApplication::resolveClientAddress() const
+{
+    QHostAddress clientAddress = m_env.clientAddress;
+
+    if (!m_isReverseProxyEnabled)
+        return clientAddress;
+
+    // Only reverse proxy can overwrite client address
+    if (clientAddress.isEqual(m_reverseProxyAddress))
+    {
+        const QString forwardedFor = m_request.headers.value(Http::HEADER_X_FORWARDED_FOR);
+
+        if (!forwardedFor.isEmpty())
+        {
+            // client address is the 1st global IP in X-Forwarded-For or, if none available, the 1st IP in the list
+            const QStringList remoteIpList = forwardedFor.split(",");
+            bool hasGlobalIp = false;
+
+            for (const QString &remoteIp : remoteIpList)
+            {
+                if (clientAddress.setAddress(remoteIp) && clientAddress.isGlobal())
+                {
+                    hasGlobalIp = true;
```

Comment:
You can `return clientAddress;` here.

Reply:
So what about changing it?

PS  I hope that your disagreements with @qix67 do not mean that this PR is not ready?

Reply:
> So what about changing it?

I've changed it in the last commit.

> PS I hope that your disagreements with @qix67 do not mean that this PR is not ready?

No we only have some differences in proxy configuration, this PR can fit both cases.